### PR TITLE
chore: don't register json serializer by default

### PR DESCRIPTION
**Pull request by fhavret**
Related to this issue https://github.com/django-money/django-money/issues/636, registering the serializer within `djmoney` package will override any customised json serializer provided by the user. The user should have the control of using the serializer provided by `djmoney` with Django `SERIALIZATION_MODULES` setting.

### CODE REVIEW

**Comment by washeck** [comment]
I think this change requires documentation update and changelog entry.

**Comment by benjaoming** [comment]
This will be a breaking change, so definitely :100: needs documentation, a changelog entry and a minor version bump.

**Comment by fhavret** [comment]
Provided documentation, bumped version and added a changelog entry 👍 
Also a side question, why providing a custom deserializer but no custom serializer ?

File: docs/changes.rst [hunk 1]
```diff
@@ -17,6 +17,13 @@ Changelog
 
 - (add entry here)
 
+`3.4` - unreleased
```
Comment (Stranger6667) [inline docs/changes.rst:20]: I would leave this change to the commit which bumps the version and issues a release

Comment (Stranger6667) [reply docs/changes.rst:20]: We have a template for this a few lines above, i.e the original intention was to fill this template (IRC)

Comment (fhavret) [reply docs/changes.rst]: Updated 👍

Comment (Stranger6667) [reply docs/changes.rst]: Thanks!

Comment (Stranger6667) [reply docs/changes.rst]: Oh, I meant that the changelog entry should still be in this PR, but placed above what it used to be in the original commit in the templated block

Comment (fhavret) [reply docs/changes.rst]: Ahh my bad ! Just updated it again

**Review by benjaoming — APPROVED** [review]
LGTM!

The one who does the next release (perhaps me? I'll write a TODO note) will notice this change in the changelog and know to bump the minor version instead of the patch version :+1:

**Review by Stranger6667 — APPROVED** [review]

**Pull request merged**